js/VimeoView.js

```diff
@@ -0,0 +1,86 @@
+define([
+    'libraries/vimeo-player-2.8.0.min'
+], function(Player) {
+    return Backbone.View.extend({
+
+        template: 'vimeoVideo',
+
+        vimeoEvents: [
+            'play',
+            'pause',
+            'ended',
+            'timeupdate',
+            'seeking',
+            'seeked',
+            'error',
+            'loaded'
+        ],
+
+        className: 'vimeo-video',
+
+        initialize: function() {
+            this.setupPlayer();
+            this.setupResponsiveSizing();
+            this.setupEventListeners();
+        },
+
+        /**
+         * Instantiate the vimeo player with the options supplied
+         */
+        setupPlayer: function() {
+            var options = {
+                url: this.model._source,
+                autoplay: this.model._autoplay,
+                loop: this.model._loop
+            };
+
+            this.player = new Player(this.el, options);
+        },
+
+        /**
+         * Use the vimeo player's methods to determine the aspect ratio,
+         * and set the padding on this view's element accordingly
+         */
+        setupResponsiveSizing: function() {
+            var $el = this.$el;
+            var player = this.player;
+            var self = this;
+
+            player
+                .getVideoWidth()
+                .then(function(width) {
```

Comment:
will this work in IE?

Reply:
Fair point, and TBH I should have checked before but looks like the Vimeo library ships with its own Promise polyfill. Tested in IE11 and it works